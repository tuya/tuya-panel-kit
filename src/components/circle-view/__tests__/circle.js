import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CircleView from '../index';

describe('CircleView Component', () => {
  it('basic render', () => {
    const component = renderer.create(
      <CircleView radius={5} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });


  it('render with background', () => {
    const component = renderer.create(
      <CircleView
        radius={25}
        color={'red'}
        borderColor="#fefefe"
        style={{ backgroundColor: 'yellow' }}
      />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

});
