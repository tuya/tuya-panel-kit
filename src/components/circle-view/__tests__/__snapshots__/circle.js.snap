// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleView Component basic render 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": null,
      },
      undefined,
      Object {
        "borderRadius": 5,
        "height": 10,
        "overflow": "hidden",
        "width": 10,
      },
    ]
  }
/>
`;

exports[`CircleView Component render with background 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "red",
        "borderColor": "#fefefe",
        "borderWidth": 1.5,
      },
      Object {
        "backgroundColor": "yellow",
      },
      Object {
        "borderRadius": 25,
        "height": 50,
        "overflow": "hidden",
        "width": 50,
      },
    ]
  }
/>
`;
